### PR TITLE
Show page

### PR DESCRIPTION
**Pull request by Vitalia888-170**
image

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "files.associations": {
+    "*.html.erb": "html"
+ }
+}
```
Comment (Shabaldas) [inline .vscode/settings.json:5]: Why do you need this file?

Comment (Vitalia888-170) [reply .vscode/settings.json:5]: I need it for html erb formatting

File: app/javascript/controllers/sidebar_controller.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { Controller } from '@hotwired/stimulus'
 
 export default class extends Controller {
-  static targets=[ "sidebarContainer", "sidebarBg" ]
+  static targets=[ "sidebarContainer"]
```
Comment (Shabaldas) [inline app/javascript/controllers/sidebar_controller.js]: ```static targets=["sidebarContainer"]```

File: app/javascript/controllers/toggle_controller.js [hunk 3]
```diff
@@ -13,16 +13,28 @@ export default class extends Controller {
     this.enteringClass = this.data.get('enteringClass') || null
     this.leavingClass = this.data.get('leavingClass') || null
 
+    if(this.hasAccordionIconTarget){
```
Comment (Shabaldas) [inline app/javascript/controllers/toggle_controller.js]: plz. add ident after ```if```

File: app/javascript/controllers/toggle_controller.js [hunk 4]
```diff
@@ -13,16 +13,28 @@ export default class extends Controller {
...
-      this.buttonTarget.addEventListener("keydown", this._onMenuButtonKeydown)
+      this.buttonTarget.addEventListener("keydown", this._onContentButtonKeydown)
     }
 
     this.element.setAttribute("aria-haspopup", "true")
   }
 
   disconnect() {
+    
+    if(this.hasAccordionIconTarget){
```
Comment (Shabaldas) [inline app/javascript/controllers/toggle_controller.js]: the same

File: app/javascript/controllers/toggle_controller.js [hunk 5]
```diff
@@ -42,20 +54,25 @@ export default class extends Controller {
   _show(cb) {
     setTimeout(
       (() => {
-        this.menuTarget.classList.remove(this.toggleClass)
+        this.contentTarget.classList.remove(this.toggleClass)
         this.element.setAttribute("aria-expanded", "true")
+        if(this.hasAccordionIconTarget){
```
Comment (Shabaldas) [inline app/javascript/controllers/toggle_controller.js]: the same

File: app/javascript/controllers/toggle_controller.js [hunk 6]
```diff
@@ -69,23 +86,28 @@ export default class extends Controller {
     setTimeout(
       (() => {
         this.element.setAttribute("aria-expanded", "false")
-        this._invisibleClassList[0].forEach(klass => this.menuTarget.classList.add(klass))
-        this._visibleClassList[0].forEach(klass => this.menuTarget.classList.remove(klass))
-        this._leavingClassList[0].forEach(klass => this.menuTarget.classList.add(klass))
+        this._invisibleClassList[0].forEach(klass => this.contentTarget.classList.add(klass))
+        this._visibleClassList[0].forEach(klass => this.contentTarget.classList.remove(klass))
+        this._leavingClassList[0].forEach(klass => this.contentTarget.classList.add(klass))
+        if(this.hasAccordionIconTarget){
```
Comment (Shabaldas) [inline app/javascript/controllers/toggle_controller.js]: the same

File: app/javascript/packs/application.js [hunk 7]
```diff
@@ -7,6 +7,7 @@ import Rails from "@rails/ujs"
 import Turbolinks from "turbolinks"
 import * as ActiveStorage from "@rails/activestorage"
 import "channels"
+import 'swiper/swiper-bundle.min.css'
```
Comment (Shabaldas) [inline app/javascript/packs/application.js]: replace ' => "

File: app/views/shared/_header.html.erb [hunk 8]
```diff
@@ -81,8 +81,8 @@
                 </svg>
               </span>
             </div>
-            <div id='activeTarget' data-dropdown-target="menu" class="absolute right-0 lg:-right-12 mt-2 hidden">
-              <div class="w-38 shadow rounded border overflow-hidden">
+            <div id='activeTarget' data-toggle-target="content" class="absolute right-0 lg:-right-12 mt-2 hidden">
+              <div class="w-38 shadow  rounded border overflow-hidden">
```
Comment (Shabaldas) [inline app/views/shared/_header.html.erb]: remove extra ident in class

File: app/javascript/controllers/sidebar_controller.js (hunk 2 above)
Comment (Vitalia888-170) [reply app/javascript/controllers/sidebar_controller.js]: fixed

File: app/javascript/controllers/toggle_controller.js (hunk 3 above)
Comment (Vitalia888-170) [reply app/javascript/controllers/toggle_controller.js]: fixed

File: app/javascript/controllers/toggle_controller.js (hunk 4 above)
Comment (Vitalia888-170) [reply app/javascript/controllers/toggle_controller.js]: fixed

File: app/javascript/controllers/toggle_controller.js (hunk 5 above)
Comment (Vitalia888-170) [reply app/javascript/controllers/toggle_controller.js]: fixed

File: app/javascript/controllers/toggle_controller.js (hunk 6 above)
Comment (Vitalia888-170) [reply app/javascript/controllers/toggle_controller.js]: fixed

File: app/javascript/packs/application.js (hunk 7 above)
Comment (Vitalia888-170) [reply app/javascript/packs/application.js]: fixed

File: app/views/shared/_header.html.erb (hunk 8 above)
Comment (Vitalia888-170) [reply app/views/shared/_header.html.erb]: fixed

**Review by Shabaldas — APPROVED** [review]

**Pull request merged**